Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -180,24 +180,26 @@ IRelocator relocator
       }
 
       var cancellationSource = new CancellationTokenSource();
-      var verifiedDocumentTask = VerifyDocumentIfRequiredAsync(databaseEntry, cancellationSource.Token);
+      var verifiedDocuments = VerifyDocumentIfRequired(databaseEntry, cancellationSource.Token);
       var updatedEntry = new DocumentEntry(
         databaseEntry.Version,
         databaseEntry.ResolvedDocument,
-        verifiedDocumentTask,
+        verifiedDocuments.Select(Task.FromResult).DefaultIfEmpty(databaseEntry.ResolvedDocument).ToTask(cancellationSource.Token).Unwrap(),
         cancellationSource
       );
       documents[documentId.Uri] = updatedEntry;
-      return verifiedDocumentTask.ToObservable();
+      return verifiedDocuments;
     }
 
-    private async Task<DafnyDocument> VerifyDocumentIfRequiredAsync(IDocumentEntry databaseEntry, CancellationToken cancellationToken) {
-      var document = await databaseEntry.ResolvedDocument;
-      if (!RequiresOnSaveVerification(document)) {
-        throw new TaskCanceledException();
-      }
-      var verifiedDocumentTask = documentLoader.VerifyAsync(document, cancellationToken);
-      return await verifiedDocumentTask;
+    private IObservable<DafnyDocument> VerifyDocumentIfRequired(IDocumentEntry databaseEntry, CancellationToken cancellationToken) {
```

Comment:
Why are we changing from a task to an observable here?  Are we returning multiple results?